src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/BuckToGeneralTestEventsConverter.java

```diff
@@ -0,0 +1,188 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.intellij.ideabuck.configurations;
+
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.BuckBuildEndConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.BuckBuildStartConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.CompilerErrorConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.TestResultsAvailableConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.TestRunCompleteConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.consumers.TestRunStartedConsumer;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.parts.TestCaseSummary;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.parts.TestResults;
+import com.facebook.buck.intellij.ideabuck.ws.buckevents.parts.TestResultsSummary;
+import com.facebook.buck.test.result.type.ResultType;
+import com.google.common.collect.ImmutableSet;
+import com.intellij.execution.process.ProcessAdapter;
+import com.intellij.execution.process.ProcessEvent;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.process.ProcessOutputTypes;
+import com.intellij.execution.testframework.TestConsoleProperties;
+import com.intellij.execution.testframework.sm.runner.GeneralTestEventsProcessor;
+import com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter;
+import com.intellij.execution.testframework.sm.runner.events.TestFailedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestFinishedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestOutputEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestStartedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestSuiteFinishedEvent;
+import com.intellij.execution.testframework.sm.runner.events.TestSuiteStartedEvent;
+import com.intellij.openapi.project.Project;
+import com.intellij.util.messages.MessageBusConnection;
+
+import org.jetbrains.annotations.NotNull;
+
+
+public class BuckToGeneralTestEventsConverter extends OutputToGeneralTestEventsConverter
+    implements TestRunStartedConsumer,
+    TestRunCompleteConsumer,
+    TestResultsAvailableConsumer,
+    BuckBuildStartConsumer,
+    BuckBuildEndConsumer,
+    CompilerErrorConsumer {
+
+  @NotNull
+  private final ProcessHandler myHandler;
+  @NotNull
+  private final Project mProject;
+  @NotNull
+  final MessageBusConnection mConnection;
+
+  public BuckToGeneralTestEventsConverter(
+      @NotNull String testFrameworkName,
+      @NotNull TestConsoleProperties consoleProperties,
+      @NotNull ProcessHandler handler,
+      @NotNull Project project) {
+    super(testFrameworkName, consoleProperties);
+    myHandler = handler;
+    mProject = project;
+    mConnection = mProject.getMessageBus()
```

Comment:
Not sure it's a good idea to connect and subscribe in the constructor. Isn't `consumeTestRunStarted` a better place for that?